internal/kinds/capp/utils/route.go

```diff
@@ -43,78 +45,78 @@ func IsDNSRecordAvailable(ctx context.Context, k8sClient client.Client, name, na
 	return available, nil
 }
 
-// GetDNSConfig returns the data of the DNS ConfigMap.
+// GetDNSConfig returns the data of the DNS for the CappConfig CRD.
 func GetDNSConfig(ctx context.Context, k8sClient client.Client) (map[string]string, error) {
-	routeCM := corev1.ConfigMap{}
-	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: CappNS, Name: dnsCM}, &routeCM); err != nil {
-		return nil, fmt.Errorf("could not fetch configMap %q from namespace %q: %w", dnsCM, CappNS, err)
+	cappConfig := v1alpha1.CappConfig{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: CappNS, Name: cappConfigName}, &cappConfig); err != nil {
+		return nil, fmt.Errorf("could not fetch cappConfig %q from namespace %q: %w", cappConfigName, CappNS, err)
 	}
 
-	return routeCM.Data, nil
+	return cappConfig.Spec.DNSConfig.Data, nil
 }
 
-// GetDNSRecordFromConfig returns the DNSRecord to be used for the record from a ConfigMap.
+// GetDNSRecordFromConfig returns the DNSRecord to be used for the record from a CappConfig CRD.
 func GetDNSRecordFromConfig(dnsConfig map[string]string) (string, error) {
 	dnsRecord := ""
 	var ok bool
 
 	if len(dnsConfig) > 0 {
 		dnsRecord, ok = dnsConfig[cnameKey]
 		if !ok {
-			return dnsRecord, fmt.Errorf("%q key is not set in ConfigMap %q", cnameKey, dnsCM)
+			return dnsRecord, fmt.Errorf("%q key is not set in CappConfig DNSConfig a", cnameKey)
 		} else if dnsRecord == "" {
-			return dnsRecord, fmt.Errorf("%q is empty in ConfigMap %q", cnameKey, dnsCM)
+			return dnsRecord, fmt.Errorf("%q is empty in CappConfig DNSConfig data", cnameKey)
 		}
 	}
 
 	return dnsRecord, nil
 }
 
-// GetZoneFromConfig returns the zone to be used for the record from a ConfigMap.
+// GetZoneFromConfig returns the zone to be used for the record from a CappConfig CRD.
 func GetZoneFromConfig(dnsConfig map[string]string) (string, error) {
```

Comment:
If we remove the `omitempty` tag from the fields in the cappconfig, this ensures that they exist. So most of this logic is not needed.